idseq_dag/steps/run_alignment_remotely.py

```diff
@@ -22,6 +22,18 @@ class PipelineStepRunAlignmentRemotely(PipelineStep):
     Run gsnap/rapsearch2 remotely
     '''
 
+    def get_input_file_validation_errors(self):
+        # Return errors if either input file is empty.
+        errors = PipelineStep.validate_input_files_min_reads(self.get_input_fas(), 1)
+
+        if errors:
+            return {
```

Comment:
I'm curious about the design behind this data structure. What if the entries of "errors" actually had multiple different error types? Shouldn't the return value be an array of dicts maybe, each entry in the array specifying one of the errors and its type?

Reply:
That's a good point.

I think I may have over-engineered this data structure. I think just returning a single error_type should be sufficient. Having the failed step and the error_type in `input_errors.json` should be sufficient for now, and you can always dig into the logs if you want to go deeper.

I want to restrict to a single error_type instead of allowing for an array of error_types to simplify things. On the front-end, we will be using the error_type to pick which user-friendly error message to display on the sample report page. I think there will always be a primary error_type. For example, INSUFFICIENT_READS or MALFORMED_INPUT_FILE.

I propose adding a log statement for any specific errors and removing the errors array. We can just return a single error_type. Let me know if this sounds good.